packages/model/addon/-private/system/promise-array-proxy.ts

```diff
@@ -0,0 +1,375 @@
+import { assert, deprecate } from '@ember/debug';
+import { dependentKeyCompat } from '@ember/object/compat';
+import { tracked } from '@glimmer/tracking';
+import Ember from 'ember';
+
+import { resolve } from 'rsvp';
+
+import {
+  DEPRECATE_EVENTED_API_USAGE,
+  DEPRECATE_RELATIONSHIP_ARRAY_PROXY_METHODS,
+} from '@ember-data/private-build-infra/deprecations';
+
+/**
+ @module @ember-data/model
+ */
+let _ExportedClass;
+/**
+  This class is returned as the result of accessing an async hasMany relationship
+  on an instance of a Model extending from `@ember-data/model`.
+
+  A PromiseManyArray is an array-like proxy that also proxies certain method calls
+  to the underlying ManyArray in addition to being "promisified".
+   
+  Right now we proxy:
+
+    * `reload()`
+    * `createRecord()`
+
+  This promise-proxy behavior is primarily to ensure that async relationship interact
+  nicely with templates. In your JS code you should resolve the promise first.
+
+  ```js
+  const comments = await post.comments;
+  ```
+
+
+
+  @class PromiseManyArray
+  @private
+*/
+class PromiseArrayProxy {
```

Comment:
doc mismatch

Reply:
docs are right, class name is wrong. Used a new class name to avoid the old one while I had both for testing. Will update.